### ISSUE TITLE
Error format

### ISSUE THREAD

**Issue by monolithed** [DJWassink/Promise-parallel-throttle#19]
```ts
try {
  let [ response ] = await PromiseThrottle.all(queue, concurrency, true);
}
catch (error) {
   console.log(error.message);
}
```

Output is empty

```ts
try {
  let [ response ] = await PromiseThrottle.all(queue, concurrency, true);
}
catch (error) {
   console.log(error);
}
```

Output is:

```
{ amountDone: 0,
  amountStarted: 1,
  amountResolved: 0,
  amountRejected: 1,
  rejectedIndexes: [ 0 ],
  resolvedIndexes: [],
  taskResults: 
   [ TypeError: Cannot read property 'reply' of undefined
         at Deliveryd.getFlags (/Users/a.abashkin/workspace/mail.ru/@qa/transport/lib/deliveryd.js:198:38)
         at Array.queue.push (/Users/a.abashkin/workspace/mail.ru/@qa/transport/lib/deliveryd.js:261:41)
         at executeTask (/Users/a.abashkin/workspace/mail.ru/@qa/transport/node_modules/promise-parallel-throttle/build/throttle.js:32:29)
         at /Users/a.abashkin/workspace/mail.ru/@qa/transport/node_modules/promise-parallel-throttle/build/throttle.js:74:21
         at process._tickCallback (internal/process/next_tick.js:109:7) ] }
```

Is it possible to standard properties like message and stack?

**Comment by DJWassink** [comment]
You mean that the result of `all` and `sync` should be a `Error` object instead of the `Result` object?

It indeed seems like I am returning the complete result object in those methods if the queue throwed. To follow the `Promise.All` spec it should maybe return a `Error` object.

**Comment by monolithed** [comment]
> You mean that the result of all and sync should be a Error object instead of the Result object?

Yep

**Comment by DJWassink** [comment]
Fix incoming, thnx!

**Comment by DJWassink** [comment]
Should be fixed in version v2.5.0 🌮

**Comment by monolithed** [comment]
Thanks!
